kotlin-react-dom/src/main/kotlin/react/dom/client/hydrateRoot.kt

```diff
@@ -0,0 +1,16 @@
+@file:JsModule("react-dom/client")
+@file:JsNonModule
+
+package react.dom.client
+
+import org.w3c.dom.Element
+import react.ReactNode
+
+// See https://reactjs.org/docs/react-dom.html
+
+// 18.0+
+external fun hydrateRoot(
+  container: Element,
+  children: ReactNode?,
```

Comment:
[Original name](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f589c8f76383f3fab94631052080ae2a81c88305/types/react-dom/client.d.ts#L51) is preferred

Reply:
@turansky fixed, and added `HydrationOptions` as well to match [this interface](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f589c8f76383f3fab94631052080ae2a81c88305/types/react-dom/client.d.ts#L8-L14), which may change over time.